pkg/mimir/mimir.go

```diff
@@ -98,6 +99,7 @@ type Config struct {
 	Target                          flagext.StringSliceCSV `yaml:"target"`
 	MultitenancyEnabled             bool                   `yaml:"multitenancy_enabled"`
 	NoAuthTenant                    string                 `yaml:"no_auth_tenant" category:"advanced"`
+	VaultEnabled                    bool                   `yaml:"vault_enabled" category:"experimental"`
```

Comment:
Can we move this inside the vault config data structure? We typically have a `enabled` config option inside the data struct config you want to enable.